nowcasting_dataset/dataset/split/method.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+
+
+def split_method(
+    datetimes: pd.DatetimeIndex,
+    train_test_validation_split: Tuple[int] = (3, 1, 1),
+    method: str = "modulo",
+    freq: str = "D",
+) -> (List[pd.Timestamp], List[pd.Timestamp], List[pd.Timestamp]):
+    """
+    Split the data by the day into train, test and validation.
+
+    method: modulo
+    If the split is (3,1,1) then, taking all the days in the dataset:
+    - train data will have all days that are modulo 5 remainder 0,1 or 2,
+    i.e 1st, 2nd, 3rd, 6th, 7th, 8th, 11th .... of the whole dataset
+    - validation data will have all days that are modulo 5 remainder 3, i.e 4th, 9th, ...
+    - test data will have all days that are modulo 5 remainder 4, i.e 5th, 10th ,  ...
+
+    method: random
+    If the split is (3,1,1) then
+    - train data will have 60% of the days
+    - validation data will have 20% of the days
+    - test data will have have 20% of the days
+
+    Args:
+        datetimes: list of datetimes
+        train_test_validation_split: how the split is made
+        method: which method to use. Can be modulo or random
+        freq: This can be D=day, W=week, M=month and Y=year. This means the data is divided up by different periods
+
+    Returns: train, validation and test datetimes
+
+    """
+
+    # find all the unique periods (dates, weeks, e.t.c)
+    datetimes_period = pd.to_datetime(datetimes.to_period(freq).to_timestamp())
+    unique_periods_in_dataset = datetimes_period.unique()
+
+    # find total weights, and cumulative weights
+    total_weights = sum(train_test_validation_split)
```

Comment:
I think I'm just being dumb (and single-minded about what 'weights' means!) but why use the word 'weights' in the variable names `total_weights` and `cum_weights`?

Reply:
yea, becasue its adding up those weights, or ratios. Not quite sure what a better name would be, but Ill have a think

Reply:
ah, cool, got it!  `weights` makes perfect sense now - thanks for explaining!